include/aws/http/request_response.h

```diff
@@ -523,6 +638,39 @@ struct aws_http_message *aws_http_message_new_request_with_headers(
 AWS_HTTP_API
 struct aws_http_message *aws_http_message_new_response(struct aws_allocator *allocator);
 
+/**
+ * xxx
+ */
+AWS_HTTP_API
+struct aws_http_message *aws_http2_message_new_request(struct aws_allocator *allocator);
+/**
+ * xxx
+ */
+AWS_HTTP_API
+struct aws_http_message *aws_http2_message_new_request_with_pseudo_headers(
+    struct aws_allocator *allocator,
+    struct aws_pseudo_headers *pseudo_headers);
+/**
+ * xxx
+ */
+AWS_HTTP_API
+struct aws_http_message *aws_http2_message_new_response(struct aws_allocator *allocator);
+
+/**
+ * Transform an HTTP/2 message to HTTP/1.1 message.
+ * Pesudo headers will be transformed to message start-line
+ * :authority will be transformed to host header.
+ */
+AWS_HTTP_API
+int aws_http2_message_to_http_message(struct aws_http_message *h2_message);
+/**
+ * Transform an HTTP/1.1 message to HTTP/2 message.
+ * Pesudo headers will be created from the context.
+ * (Maybe more, connection-specific header will be removed)
+ */
+AWS_HTTP_API
+int aws_http_message_to_http2_message(struct aws_http_message *h2_message);
+
 /**
  * Acquire a hold on the object, preventing it from being deleted until
```

Comment:
If we're going to have a distinct aws_http_pseudo_headers class, we'll need  _get_pseudo_headers() and _get_const_pseudo_headers() functions